This warning is not related to my code. 
I cross checked this again by setting up a new CMSSW area in 8_1_0_pre11 and checkout the package: 
DQM/SiPixelMonitorCluster/

After compiling with @dmitrijus command and I get the same warning:
 /afs/cern.ch/work/g/gouskos/private/Tracker_2015/test/CMSSW_8_1_0_pre11/src/DQM/SiPixelMonitorCluster/interface/SiPixelClusterSource.h:67:21: warning: 'analyze' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
       virtual void analyze(const edm::Event&, const edm::EventSetup&);

--- a/DQM/SiPixelMonitorCluster/interface/SiPixelClusterSource.h
+++ b/DQM/SiPixelMonitorCluster/interface/SiPixelClusterSource.h
@@ -64,8 +64,8 @@
 
        typedef edmNew::DetSet<SiPixelCluster>::const_iterator    ClusterIterator;
        
-       virtual void analyze(const edm::Event&, const edm::EventSetup&);
-       virtual void dqmBeginRun(const edm::Run&, edm::EventSetup const&) ;
+       virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+       virtual void dqmBeginRun(const edm::Run&, edm::EventSetup const&) override;
        virtual void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
 
        virtual void buildStructure(edm::EventSetup const&);

--- a/DQM/SiPixelMonitorTrack/interface/SiPixelHitEfficiencySource.h
+++ b/DQM/SiPixelMonitorTrack/interface/SiPixelHitEfficiencySource.h
@@ -46,9 +46,9 @@ class SiPixelHitEfficiencySource : public DQMEDAnalyzer {
     explicit SiPixelHitEfficiencySource(const edm::ParameterSet&);
             ~SiPixelHitEfficiencySource();
 
-    virtual void dqmBeginRun(const edm::Run& r, edm::EventSetup const& iSetup);
+    virtual void dqmBeginRun(const edm::Run& r, edm::EventSetup const& iSetup) override;
     virtual void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
-    virtual void analyze(const edm::Event&, const edm::EventSetup&);
+    virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
     virtual void fillClusterProbability(int , int, bool, double );
 
   private: 

--- a/DQM/SiPixelMonitorTrack/interface/SiPixelTrackResidualSource.h
+++ b/DQM/SiPixelMonitorTrack/interface/SiPixelTrackResidualSource.h
@@ -51,9 +51,9 @@ class SiPixelTrackResidualSource : public DQMEDAnalyzer {
     explicit SiPixelTrackResidualSource(const edm::ParameterSet&);
             ~SiPixelTrackResidualSource();
 
-    virtual void dqmBeginRun(const edm::Run& r, edm::EventSetup const& iSetup);
+    virtual void dqmBeginRun(const edm::Run& r, edm::EventSetup const& iSetup) override;
     virtual void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
-    virtual void analyze(const edm::Event&, const edm::EventSetup&);
+    virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
     void getrococcupancy(DetId detId,const edm::DetSetVector<PixelDigi> diginp,const TrackerTopology* const tTopo,std::vector<MonitorElement*> meinput);
     void triplets(double x1,double y1,double z1,double x2,double y2,double z2,double x3,double y3,double z3,
                   double ptsig, double & dc,double & dz, double kap); 

